ValveKeyValue/ValveKeyValue/Deserialization/KVObjectBuilder.cs

```diff
@@ -17,6 +17,10 @@ public KVObject GetObject()
 
             foreach (var associatedBuilder in associatedBuilders)
             {
+                if (associatedBuilder.stateStack.Count > 0) // The #base or #include file may be empty.
+                {
+                    associatedBuilder.GetObject(); // Recurses through #base and #include directives, finalizing them all.
```

Comment:
It seems wrong to me to call `GetObject` without using the result, just to trigger side-effects that probably should be handled by `FinalizeState()`.

Reply:
I am surprised this even works, what actually makes it work? Is it because of recursive FinalizeState calls?

Reply:
I've pushed a change which avoids using `GetObject` in this way and instead handles the associated builders in `FinalizeState`. Is this satisfactory?